core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -546,7 +546,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       val appsToRetain = new mutable.LinkedHashMap[String, FsApplicationHistoryInfo]()
 
       def shouldClean(attempt: FsApplicationAttemptInfo): Boolean = {
-        now - attempt.lastUpdated > maxAge && attempt.completed
+        now - attempt.lastUpdated > maxAge
```

Comment:
Hi @vanzin , as far as I understand you are not worried about the 2+ seconds granularity but instead worried that for inprogress jobs `lastUpdated` field might never get updated based on below comment:

> some filesystems do not appear to update the `modtime` value whenever data is flushed to an open file output stream.

Do we know that above statement is true? If that's the case, your suggestion (using last loaded time) seems like proper way of solving the issue.